### ISSUE TITLE
`VPi` case of `eraseType`  appears not to handle dependent types

### ISSUE THREAD

**Issue by rokopt** [anoma/juvix#988]
The `IR.VPi` case of `eraseType` in `Erased/Algorithm.hs` contains the following:

```haskell
eraseType (IR.VPi π a b) = do
  if π == mempty
    then eraseType b
    else -- FIXME dependency
```

(There's also a 'FIXME' in `eraseElim` above that, though I don't think that one has to do with dependent types specifically -- it's the `Core.Pattern` case being just a `throwEra`.)

**Comment by andy-morris** [comment]
yeah, you're completely right. this is the reason having separate types/terms in erasure is a lie. :p

some time soon we're going to have to refactor it to have just a single syntax for both like with previous phases
